src/dbg/formatfunctions.cpp

```diff
@@ -38,11 +38,53 @@ void FormatFunctions::Init()
     {
         std::vector<wchar_t> helpMessage(destCount);
         String errName = ErrorCodeToName((unsigned int)code);
+#ifdef _WIN64
+        if((code >> 32) != 0)  //Data in high part: not an error code
+        {
+            errName = StringUtils::sprintf("%p", code);
+            if(destCount < errName.size() + 1)
+                return FORMAT_BUFFER_TOO_SMALL;
+            else
+            {
+                memcpy(dest, errName.c_str(), errName.size() + 1);
+                return FORMAT_SUCCESS;
+            }
+        }
+#endif //_WIN64
         if(errName.size() == 0)
             errName = StringUtils::sprintf("%08X", DWORD(code));
+        DWORD success = FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS, NULL, DWORD(code), MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), helpMessage.data(), DWORD(destCount), NULL);
+        if(success > 0)
+        {
+            String UTF8ErrorMessage = StringUtils::Utf16ToUtf8(helpMessage.data());
+            if(destCount < errName.size() + 3 + UTF8ErrorMessage.size())
+                return FORMAT_BUFFER_TOO_SMALL;
+            else
+            {
+                sprintf_s(dest, destCount, "%s: %s", errName.c_str(), UTF8ErrorMessage.c_str());
+                return FORMAT_SUCCESS;
+            }
+        }
+        else
+        {
+            if(destCount < errName.size() + 1)
+                return FORMAT_BUFFER_TOO_SMALL;
+            else
+            {
+                memcpy(dest, errName.c_str(), errName.size() + 1);
+                return FORMAT_SUCCESS;
+            }
+        }
+    });
+
+    Register("ntstatus", [](char* dest, size_t destCount, int argc, char* argv[], duint code, void* userdata)
+    {
+        std::vector<wchar_t> helpMessage(destCount);
+        String errName = ErrorCodeToName((unsigned int)code);
 #ifdef _WIN64
         if((code >> 32) != 0)  //Data in high part: not an error code

```

Comment:
The high part of NTSTATUS codes are usually set to something like 0xFFFFFFFFC0000005 so this check would say the code is invalid while it is actually valid...